src/core/agent/services/ipexCommunicationService.ts

```diff
@@ -540,12 +536,30 @@ class IpexCommunicationService extends AgentService {
       `${ConfigurationService.env.keri.credentials.testServer.urlInt}/oobi/${schemaSaid}`
     );
     const schema = await this.props.signifyClient.schemas().get(schemaSaid);
-    await this.ipexMessageStorage.createIpexMessageRecord({
+
+    let prefix;
+    let key;
+    switch (historyType) {
+    case ConnectionHistoryType.CREDENTIAL_REVOKED:
+      prefix = KeriaContactKeyPrefix.HISTORY_REVOKE;
+      key = message.exn.e.acdc.d;
+      break;
+    default:
+      prefix = KeriaContactKeyPrefix.HISTORY_IPEX;
+      key = message.exn.d;
+      break;
+    }
+    const ipexHistory = {
       id: message.exn.d,
       credentialType: schema?.title,
       content: message,
       connectionId: message.exn.i,
       historyType,
+      createdAt: new Date(),
```

Comment:
again this should match the interface and not the record type as its unnecessary